### ISSUE TITLE
[v1.14] sam_e70_xplained:tests/drivers/watchdog/wdt_basic_api failed with v1.14 branch.

### ISSUE THREAD

**Issue by chen-png** [zephyrproject-rtos/zephyr#23149]
**Describe the bug**
Running tests/drivers/watchdog/wdt_basic_api test case on sam_e70_xplained board with v1.14 branch, it will loop to report the message below.

This test has been skipped on sam_e70_xplained board in master branch.

**To Reproduce**
1. west build -b=sam_e70_xplained  tests/drivers/watchdog/wdt_basic_api 
2. west flash
3. See error

**Screenshots or console output**
```
Running test suite wdt_basic_test

starting test - test_wdt
Testcase: test_wdt_no_callback
Waiting to restart MCU
Running test suite wdt_basic_test

starting test - test_wdt
Testcase: test_wdt_no_callback
Waiting to restart MCU
Running test suite wdt_basic_test

starting test - test_wdt
Testcase: test_wdt_no_callback
Waiting to restart MCU
Running test suite wdt_basic_test

starting test - test_wdt
Testcase: test_wdt_no_callback
Waiting to restart MCU
```

**Environment (please complete the following information):**
 - OS: Fedora28
 - Toolchain: zephyr-sdk-0.10.3
 - Commit ID: ff682926

**Comment by chen-png** [comment]
it's still an issue, reopened it

**Comment by carlescufi** [comment]
We are going to fix this for 1.14. If the bug exists in master, we will fix it there.
